### DISCUSSION TITLE
Creation of new labels to classify the issues

### DISCUSSION THREAD

**Discussion by andre-hohmann** [kitodo/kitodo-production#5564] [Q&A] [answered]
### Problem
I had a look on several issues to think about helpful improvements with regard on the development fund 2022. This is very time consuming, because nearly every issue has to be assessed on its own regarding the concerned subject areas and thus its relevance for an improvement. 

### Solution
From my point of view, the Labels should be extended to some more broad subject-categories as for example: 
- performance
- search/filter
- newspaper
- import
- ruleset
- ...

On the first glance, it might seem unnecessary, but with 470 open issues, it is from my point of view an opportunity to assess quickly the issues and to bundle them. Especially regarding the development fund but also for other development-initiatives, it would be helpful to enable a quick overview of related issues. 

It would be OK, if these labels have always be to combined with the improvement-label or with the feature-label. It just would be helpful not to assess each of the 470 issues by content.

**Comment by matthias-ronge** [comment]
For the same reason, I once started keeping [a list in the wiki](/kitodo/kitodo-production/wiki/Bugs-by-severity) to organize the issues better. I made up the following categories, in order of decreasing **priority**:

- **Dangerous**
_A risk of life or limb, property damage, security issue or the risk of loss of work._
This is basically equal to the <kbd>security</kbd> label that we already have.
- **Unusable**
_An_ essential _functionality of the application is not given. There is_ no work-around.
This is basically equal to the <kbd>blocking</kbd> label that we already have.
- **Faulty**
_A target functionality is not given or behaves wrong._
This is what a <kbd>bug</kbd> describes best.
- **Fragile**
_An error is badly intercepted. A tiny incorrect operation leads to incredibly bad things._
- **Unsightly**
_Slipped layout, typographical errors, …_
- **Inconvenient**
_A target functionality is not available, but_ there is a work-around. _Likewise: A target functionality can only be achieved with an unnecessarily high operating effort._
- **At a loss**
_Poor operator guidance, the meaning of input elements is not revealed, nothing seems to happen when a function is triggered (if the function actually works). Likewise: No error message in the event of an error, it gives the impression that the function has worked where it did not._
- **Opaque program code**
_Confusing program code, cleaning up necessary._
- **Non-interoperable software solution**
_An in-house solution is used where a standard solution could be used._
- **New feature**
_A completely new type of functionality is desired._
This exists as <kbd>feature</kbd> label
- **Conception**
_Discussion with an open outcome, no clear goal yet._
These should be moved to [discussions](/kitodo/kitodo-production/discussions)

Aside from that, I came up with the following **subject keywords**:

- administrability
- business objects
- calendar hierarchies
- <kbd>documentation</kbd>
- export
- hierarchical processes
- import
- layout
- list view
- metadata editor
- migration
- multitenancy
- performance
- process creation
- search
- <kbd>security</kbd>
- statistics
- tasks
- text
- workflow editor
- workflow engine

**Comment by andre-hohmann** [comment]
@matthias-ronge : Thanks, that is very close to what i expected. The creation of your list was certainly quite time consuming. From my point of view, that should be solved with the labels. 

The priority labels i had not in mind, but maybe the existing ones could be complemented. Regarding the subject labels, i prefer the following ones: 
- calendar hierarchies
- export
- import
- layout
- metadata editor
- migration
- performance
- search

Could we create and apply them to have a look how it works? Is there someone who organizes the labels and decides about new ones? 
I suggest to use the same color for all subject labels to indicate the subject-reference. Maybe a neutral color as a white background would be good to show that the subject label is a supplement to the colorful "priority" labels.

**Comment by matthias-ronge** [comment]
I don't have permission to do this

**Comment by matthias-ronge** [comment]
@Kathrin-Huber @Erikmitk @henning-gerhardt What do you think about this? Could one of you perhaps create a few more meaningful labels?

**Comment by henning-gerhardt** [comment]
I'm not the right person for this @matthias-ronge as I'm not an administrator nor release manager.

**Comment by andre-hohmann** [comment]
Here a part of the list from: [Bugs by severity](https://github.com/kitodo/kitodo-production/wiki/Bugs-by-severity) with the current state. The aim to copy it here is to see, which issues are still open - especially the important ones, 

### 10 Dangerous
- #4106  `security`
- #456 `security`

### 9 Unusable
- #3190 `migration`
- #3275 `metadata editor` `hierarchical processes`
- #3368 `import` `export`
- #3569 `migration`
- #3713 `search`
- #3722 <kbd>calendar hierarchies</kbd>
- #3741 <kbd>calendar hierarchies</kbd>
- #3751 <kbd>metadata editor</kbd>
- #3762 <kbd>metadata editor</kbd>
- #4064 <kbd>export</kbd>
- #4068 <kbd>metadata editor</kbd>
- #4069 <kbd>metadata editor</kbd>

### 8 Faulty
_A target functionality is not given or behaves wrong._

- #3156 <kbd>tasks</kbd>
- #3155 <kbd>tasks</kbd>
- #3292 <kbd>list view</kbd>
- #3341 <kbd>search</kbd>
- #3403 <kbd>process creation</kbd>
- #3408 <kbd>process creation</kbd>
- #3486 <kbd>metadata editor</kbd>
- #3512 <kbd>search</kbd>
- #3523 <kbd>search</kbd>
- #3586 <kbd>administrability</kbd>
- #3640 <kbd>import</kbd>
- #3646 <kbd>metadata editor</kbd>
- #3649 <kbd>business objects</kbd>
- #3664 <kbd>process creation</kbd>
- #3672 <kbd>workflow engine</kbd>
- #3673 <kbd>workflow engine</kbd>
- #3692 <kbd>business objects</kbd>
- #3718 <kbd>workflow engine</kbd>
- #3800 <kbd>import</kbd>
- #3812 <kbd>calendar hierarchies</kbd>
- #3818 <kbd>workflow engine</kbd>
- #3880 <kbd>process creation</kbd>
- #3938 <kbd>search</kbd>
- #4017 <kbd>metadata editor</kbd>
- #4099 <kbd>statistics</kbd>
- #4129 <kbd>session management</kbd>
- #4150<kbd>business objects</kbd>
- #4152 <kbd>business objects</kbd>
- #4155 <kbd>metadata editor</kbd>

**Comment by andre-hohmann** [comment, marked as answer]
I moved this issue to the discussion-area, as it is not really an issue. 

With the new labels added on 2023-02-24, the question is answered.
